shell/plugins/steve/subscribe.js

```diff
@@ -944,19 +970,7 @@ const defaultGetters = {
     if ( !revision ) {
       const cache = state.types[type];
 
-      if ( !cache ) {
```

Comment:
Had a quick chat with @KevinJoiner just to see what the correct approach would be from a backend perspective would be and I was able to confirm that the Collection Revision just isn't really related to any of the Resource Revisions that we'd find on resources inside of the collection so we shouldn't be pulling a revision off of a resource inside the collection to send a watch request for the Resource Collection.

I think the correct approach here should be that if we're ever attempting to watch a Resource Collection and we don't have a revision (and it's not a special exception like schema and a few others) then we should always just re-perform the GET request on the collection and use the revision from that response to watch the collection.

Reply:
> I think the correct approach here should be that if we're ever attempting to watch a Resource Collection and we don't have a revision (and it's not a special exception like schema and a few others) then we should always just re-perform the GET request on the collection and use the revision from that response to watch the collection.

Agreed, here are some docs for how K8s designed the desired workflow. 
https://kubernetes.io/docs/reference/using-api/api-concepts/#efficient-detection-of-changes
> When the requested watch operations fail because the historical version of that resource is not available, clients must handle the case by recognizing the status code 410 Gone, clearing their local cache, performing a new get or list operation, and starting the watch from the resourceVersion that was returned.

Reply:
I think there's a few confused points here. We should resolve this in a call when I'm back. 

Generally, falling back on a resource revision has worked extremely well and when we tried to improve that we hit a lot of bugs. We should also avoid re-fetching if we can, beside 'too old'. Given how many times we receive resource.stop messages and the amount of resources we could be fetching there's major scaling issues

Reply:
The scope of this PR is to enable the advanced worker for management and rancher stores. Changing how we handle socket functionality outside of the advanced worker isn't in scope. For reliability reasons the advanced worker should mirror that current functionality which works along the following principles (ignoring the concept of higher/lower revision numbers, which hasn't impacted us)...

1) we fetch a collection of resources that returns a resource revision at that current point in history
2) we receive updates to those resources each containing a newer resource revision. if we were to fetch the collection of resources at the exact same time we received an update the resource revisions would be matching
3) subscribing to a resource type with a resource revision from an individual resource should then mean we're at the latest known point in our cache and should get any events following it

This process has been relied upon in the dashboard for years. The only real issue we've had was when upon a `too old` error the same too old revision was used to re-sub, that's now been handled (we block the watch request and wait for a full resync/http fetch). 

If we need to update this process, it should be done very carefully at a later date.

Let's run through this in the call later on.

Reply:
Gonna revert this for now but I'm uncomfortable with this approach because by all rights it shouldn't work (and potentially doesn't except by accident) but it doesn't fail in a way that's easy to discern and if it does succeed on a subscription, we're almost certain to get a lot of excess messages/mutations for the collection revision subscribed to catches up with the actual current collection state.